# Description of configuration files can be found at "https://gitlab.ethz.ch/ics-group/projects/andrea/crs-2.0/-/wikis/ros-configuration"
state_type: pacejka_car
input_type: pacejka_car

controller_type: FF_FB
max_rate: 50
visualization_rate: 10

controller_params:
  target_velocity: 2.0
  use_filter: true
  b_filter: [0.0985,0.2956,0.2956,0.0985]
  a_filter: [1.0000,-0.5772,0.4218,-0.0563]

  Kd: 0.01
  Kp: 3.0
  Ki: 0.0

  # crs uses a torque between 0 and 1. The kinematic model uses a different torque range internally.
  # the conversion is done with the linear function torque_km = a * torque_crs + b
  a_torque:  6.44026018
  b_torque:   0.13732343
  steer_limit: 0.4

  K_torque_curv: 0.32
  mean_curv_dist: 70 # in number of track indices
  look_ahead_dist: 0.15
  lag_compensation_time: 0.2




visualizer:
  rate: 10
  type: last_reference
  # Visualizer specific  parameters
  frame_id: crs_frame # default
  namespace: ff_fb # default

  use_arrows: false # If true, use arrows to visualize planned and reference yaw angle. This option is a lot slower and may introduce visual lags


  r: 1
  g: 0
  b: 0
  a: 1
  size_x: 0.05 # 0.07 for arrows
  size_y: 0.05 # 0.02 for arrows
  size_z: 0.05 # 0.02 for arrows
